compiler/rustc_session/src/session.rs

```diff
@@ -365,14 +364,6 @@ impl Session {
     pub fn struct_span_warn<S: Into<MultiSpan>>(&self, sp: S, msg: &str) -> DiagnosticBuilder<'_> {
         self.diagnostic().struct_span_warn(sp, msg)
     }
-    pub fn struct_span_warn_with_code<S: Into<MultiSpan>>(
```

Comment:
cc @rust-lang/wg-diagnostics this PR removes lots of unused APIs... I personally think we should keep it, so that the session exposes the entire API of the diagnostic system, but we could just go through the `diagnostic()` method to get the same API back.

Reply:
+1 for keeping, I'd consider as "we just don't use it for now".

Reply:
The diagnostic system has a *lot* of public functions, which are exposed in three places now:
1. `Diagnostic`
2. `DiagnosticBuilder`
3. `Session`

Do we really need all three?

Reply:
probably not. But we should figure out how to move forward to something better before we make the APIs inconsistent between the three sites.